doc/versioning.md

```diff
@@ -0,0 +1,41 @@
+# gRPC Versioning Guide
+
+## Versioning Overview
+
+All gRPC implementations use a three-part version number (`vX.Y.Z`) and strictly follow [semantic versioning](https://semver.org/), which defines the semantics of major, minor and patch components of the version number. In addition to that, gRPC versions evolve according to these rules:
+- **Major version bumps** only happen on rare occasions. In order to qualify for a major version bump, certain criteria described later in this document need to be met. Most importantly, a major version increase must not break wire compatibility with other gRPC implementations so that existing gRPC libraries remain fully interoperable. 
+- **Minor version bumps** happen approx. every 6 weeks as part of the normal release cycle as defined by the gRPC release process. A new release branch named vMAJOR.MINOR.PATCH) is cut every 6 weeks based on the [release schedule](https://github.com/grpc/grpc/blob/master/doc/grpc_release_schedule.md).
+- **Patch version bump** corresponds to bugfixes done on release branch.
+
+There are also a few extra rules regarding adding new gRPC implementations (e.g. adding support for a new language)
+- New implementations start at v0.x.y version they reaches 1.0, they are considered not ready for production workloads. Breaking API changes are allowed in the 0.x releases as the library is not considered stable yet.
+- The "1.0" release has semantics of GA (generally available) and being production ready. Requirements to reach this milestone are at least these
+  - basic RPC features are feature complete and tested
+  - implementation is tested for interoperability with other languages
+  - Public API is declared stable
+- Once a gRPC library reaches 1.0 (or higher version), the normal rules for versioning apply.
+
+## Policy for updating the major version number
+
+To avoid user confusion and simplify reasoning, the gRPC releases in different languages try to stay synchronized in terms of major and minor version (all languages follow the same release schedule). Nevertheless, because we also strictly follow semantic versioning, there are circumstances in which a gRPC implementation needs to break the version synchronicity and do a major version bump independently of other languages.
+
```

Comment:
Does the minor ver remain in sync with other major versions? For example, if a lang bumps it's major version for the next release which is say 1.23.0. Does it become 2.23.0 or 2.0.0? I would think 2.23.0 but better to clarify this.